src/Umbraco.Infrastructure/PropertyEditors/ValueConverters/NestedContentSingleValueConverter.cs

```diff
@@ -37,9 +35,10 @@ public override bool IsConverter(IPublishedPropertyType propertyType)
         public override Type GetPropertyValueType(IPublishedPropertyType propertyType)
         {
             var contentTypes = propertyType.DataType.ConfigurationAs<NestedContentConfiguration>().ContentTypes;
-            return contentTypes.Length > 1
-                ? typeof(IPublishedElement)
-                : ModelType.For(contentTypes[0].Alias);
+
+            return contentTypes.Length == 1
+                ? ModelType.For(contentTypes[0].Alias)
+                : typeof(IPublishedElement);
```

Comment:
This could previously throw an `IndexOutOfRangeException` if there were no configured content types (although you might then have bigger issues when that's the case)...